src/Symfony/Component/Validator/ConstraintValidatorInterface.php

```diff
@@ -30,6 +33,10 @@ public function initialize(ExecutionContextInterface $context);
      *
      * @param mixed      $value      The value that should be validated
      * @param Constraint $constraint The constraint for the validation
+     *
+     * @throws UnexpectedTypeException       If the constraint or the value is incompatible (causes an unrecoverable error)
+     * @throws ConstraintDefinitionException If a constraint option is incompatible during runtime
```

Comment:
I am not convinced about adding this one. I had a look at the places where this exception is thrown and in my opinion that should not happen at runtime but the check should be performed in the constraint instead.